tests/foreman/api/test_reporttemplates.py

```diff
@@ -369,22 +371,78 @@ def test_negative_create_report_without_name():
 
 
 @pytest.mark.tier2
-@pytest.mark.stubbed
-def test_positive_applied_errata():
+@pytest.mark.rhel_ver_match(r'^(?!6$)\d+$')
+@pytest.mark.no_containers
+def test_positive_applied_errata(
+    module_org, module_location, module_cv, module_lce, rhel_contenthost, target_sat
+):
     """Generate an Applied Errata report
 
     :id: a4b577db-141e-4871-a42e-e93887464986
 
-    :setup: User with reporting access rights, some host with applied errata
+    :setup: A Host with some applied errata.
 
     :steps:
 
-        1. POST /api/report_templates/:id/generate
+        1. Generate an Applied Errata report
 
     :expectedresults: A report is generated with all applied errata listed
 
     :CaseImportance: Medium
     """
+    activation_key = target_sat.api.ActivationKey(
+        environment=module_lce, organization=module_org
+    ).create()
+    ERRATUM_ID = str(settings.repos.yum_6.errata[2])
+    target_sat.cli_factory.setup_org_for_a_custom_repo(
+        {
+            'url': settings.repos.yum_9.url,
+            'organization-id': module_org.id,
+            'content-view-id': module_cv.id,
+            'lifecycle-environment-id': module_lce.id,
+            'activationkey-id': activation_key.id,
+        }
+    )
+    result = rhel_contenthost.register(module_org, module_location, activation_key.name, target_sat)
+    assert f'The registered system name is: {rhel_contenthost.hostname}' in result.stdout
+    assert rhel_contenthost.subscribed
+    result = rhel_contenthost.run(f'yum install -y {FAKE_1_CUSTOM_PACKAGE}')
+    assert result.status == 0
+    result = rhel_contenthost.run(f'rpm -q {FAKE_1_CUSTOM_PACKAGE}')
+    assert result.status == 0
+    rhel_contenthost.add_rex_key(satellite=target_sat)
+    task_id = target_sat.api.JobInvocation().run(
+        data={
+            'feature': 'katello_errata_install',
+            'inputs': {'errata': ERRATUM_ID},
+            'targeting_type': 'static_query',
+            'search_query': f'name = {rhel_contenthost.hostname}',
+            'organization_id': module_org.id,
+        },
+    )['id']
+    target_sat.wait_for_tasks(
+        search_query=(f'label = Actions::RemoteExecution::RunHostsJob and id = {task_id}'),
+        search_rate=15,
+        max_tries=10,
+    )
+    rt = (
+        target_sat.api.ReportTemplate()
+        .search(query={'search': 'name="Host - Applied Errata"'})[0]
+        .read()
+    )
+    res = rt.generate(
+        data={
+            'organization_id': module_org.id,
+            'report_format': "json",
+            'input_values': {
+                'Filter Errata Type': 'all',
+                'Include Last Reboot': 'no',
+                'Status': 'all',
+            },
+        }
+    )
+    assert res[0]['erratum_id'] == ERRATUM_ID
+    assert res[0]['issued'] == '2012-01-27'
```

Comment:
I think checking the erratum ID should be enough because there are chances that the package would be replaced in future without keeping in mind that we are also asserting the issue date.

Reply:
Checking the issue date has to do with this being a newly added field in 6.14, and this is testing that feature: https://issues.redhat.com/browse/SAT-17071

Reply:
I could potentially just assert the existence of the field, but I don't think there's too much harm in checking the content as well.

Reply:
Yeah, testing the field value should be enough instead of the exact value as that would likely fail tomorrow as we change things in lab and regenerate things like RPMs etc.

Reply:
@jyejare I did not quite get your arguments. Nothing would fail tomorrow as we do not "change things in lab". We also do not "regenerate things like RPMs". And if we do, I would expect to update the constants/config in our test suite to reflect it.

@sambible It is entirely up to you, but if you plan to keep the date there, please put it in `settings.repos.yum_6.errata`.

Ideal would be 
```yaml
  YUM_6:
    URL: '@format {this[robottelo].repos_hosting_url}/needed_errata'
    ERRATA:
    - ID: RHBA-2012:1030
      DATE_ISSUED: 
    - RHEA-2012:0058
    - RHSA-2012:0055
    - RHSA-2012:0056
    - RHSA-2012:0057
```
but that would break in all places where we use errata IDs from YUM_6.

I would go with the following instead.
```yaml
  YUM_6:
    URL: '@format {this[robottelo].repos_hosting_url}/needed_errata'
    ERRATA:
    - RHBA-2012:1030
    - RHEA-2012:0058
    - RHSA-2012:0055
    - RHSA-2012:0056
    - RHSA-2012:0057
    DATE_ISSUED:
    - YYYY-MM-DD
    - YYYY-MM-DD
    - 2012-01-27
    - YYYY-MM-DD
    - YYYY-MM-DD
```

Reply:
@ogajduse  Forecasting the future is really hard :)

I wont say that we will change or wont change but lets make sure that we dont break things for such changes which are volitile especially if those are not critical. Like checking the date here is not the critical assertion of this test.

Hence I dont agree with your comment ` did not quite get your arguments. Nothing would fail tomorrow as we do not "change things in lab". We also do not "regenerate things like RPMs".`

I am not 100% agree on keeping these dates as part of `configuration` too as they dont seems to be critical. Lets not unnecessarily assert things that are not tied to the test expectation or goal and fail the entire test on it.